### ISSUE TITLE
jest-dom/prefer-checked applies to plain object

### ISSUE THREAD

**Issue by pustovalov** [testing-library/eslint-plugin-jest-dom#47]
- `eslint-plugin-jest-dom` version: 2.1.0
- `node` version: 14.2.0
- `yarn` version: 1.22.4

Relevant code or config

```javascript
const item = {"activates_at": "2020-05-14T02:05:00.000-07:00", "active": true, "checked": false}
expect(item.checked).toBeFalsy()
```

What happened:

rule `jest-dom/prefer-checked` changed code to:
```
expect(item).not.toBeChecked()
```

```
received value must be an HTMLElement or an SVGElement.
    Received has type:  object
    Received has value: {"activates_at": "2020-05-14T02:05:00.000-07:00",
```

Is this expected behavior?

**Comment by benmonro** [comment]
@pustovalov yes. This was a trade-off?  Unfortunately there's no way to statically determine if it is a Dom element or not.  I recommend using eslint disable for this.

**Comment by pustovalov** [comment]
@benmonro maybe need to change this one?
https://github.com/testing-library/eslint-plugin-jest-dom#supported-rules

`indicates that a rule is recommended for all users.`

**Comment by benmonro** [comment]
@pustovalov i thought about that.  In fact I ran into this issue myself but i still believe the role would be recommended since jest will fail if the element is not a Dom element.  You might consider opening a ticket on jest Dom top remove that part of the matcher...
